### PR TITLE
feat(#major): Mummy finance subgraph; add Mummy Finance subgraph

### PR DESCRIPTION
**Pull request by shashwatS22**
Test Deployment Links:

Mummy Finance
1. Optimism: https://thegraph.com/hosted-service/subgraph/shashwats22/mummy-finance-optimism?version=pending
2. Fantom: https://thegraph.com/hosted-service/subgraph/shashwats22/mummy-finance

GMX 
1. Arbitrum: https://thegraph.com/hosted-service/subgraph/shashwats22/gmx-arbitrum
2. Avalanche: https://thegraph.com/hosted-service/subgraph/shashwats22/gmx-avalanche

Note: Still working on updateFundingRate event issue, on adding the event( precisely on loading pool in that event) dailySnapshots stops working for the pool.

### CODE REVIEW

**Comment by dmelotik** [comment]
Hey @shashwatS22 a few things:
- Can you fix the lint errors?
- Could you add the deployment links to the gmx subgraphs as well?
     - If you add the deployment links like this (https://okgraph.xyz/?q=shashwats22%2Fmummy-finance-optimism) then no matter how many times you deploy them they will be pointed to the same subgraph. by using the deployment ID you will have to update that link each time you deploy the subgraph
- Can you make the `protocol.name` in mummy `Mummy Finance` and not `MummyFinance`?

**Comment by dmelotik** [comment]
@shashwatS22 since this also updates gmx to 1.3.1 we should make sure to update the schema and subgraph version for gmx as well.

I also want to make sure you've gone through this pr (#2129) to ensure the new fields are accounted for in this PR

**Comment by shashwatS22** [comment]
> @shashwatS22 since this also updates gmx to 1.3.1 we should make sure to update the schema and subgraph version for gmx as well.
> 
> I also want to make sure you've gone through this pr (#2129) to ensure the new fields are accounted for in this PR

I have gone through the PR (#2129 ) but I'll double check to make sure I am not missing anything and let you know.

**Comment by dmelotik** [comment]
@shashwatS22 another rebase and merge for you with this. No need to increment subgraph version on this, just updating the schema version: https://github.com/messari/subgraphs/pull/2159

**Comment by shashwatS22** [comment]
> @shashwatS22 another rebase and merge for you with this. No need to increment subgraph version on this, just updating the schema version: #2159

I'll add this and let you know, I am in the middle of doing the QA fixes, and will push these changes with QA fixes. Thanks!

**Comment by ishraq8** [comment]
QA passed, this is ready to deploy after code review.

File: subgraphs/gmx-forks/schema.graphql
Comment (dmelotik) [inline subgraphs/gmx-forks/schema.graphql:1]: Apologies for making you do so many updates after building, but can you copy over the latest schema in the gmx folder? That is the only action. Not new fields or entities

File: subgraphs/gmx-forks/src/modules/token.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    const symbol = utils.readValue(tokenContract.try_symbol(), "");
+    const decimals = utils
+      .readValue(tokenContract.try_decimals(), constants.BIGINT_ZERO)
+      .toI32();
+    return { name, symbol, decimals };
+  }
+}
+
+//if token price is zero
+// if token is GMX or GLP
```
Comment (dmelotik) [inline subgraphs/gmx-forks/src/modules/token.ts:26]: What do these comments mean here?

File: deployment/deployment.json [hunk 2]
```diff
@@ -2635,6 +2635,58 @@
...
+        "options": {
+          "prepare:yaml": true,
+          "prepare:constants": true
+        }
+      },
+      "mummy-finance-optimism": {
+        "network": "optimism",
+        "status": "dev",
+        "versions": {
+          "schema": "1.3.1",
```
Comment (dmelotik) [inline deployment/deployment.json:2670]: 1.3.2

**Review by dmelotik — COMMENTED** [review]
Nice work @shashwatS22 !

A comment:
- We cannot change the protocol or market IDs on the gmx subgraph. Since it is already prod that would break assumptions we make in ingestion. That will need to be fixed.
- Also, the gmx subgraphs still will take over a month to sync. Is there anyway we can do something similar to what is done on the current gmx subgraph to speed it up? I know we have talked about this before, and I don't remember what our resolution was.

File: subgraphs/gmx-forks/schema.graphql
Comment (shashwatS22) [reply subgraphs/gmx-forks/schema.graphql:1]: Done

File: subgraphs/gmx-forks/src/modules/token.ts (hunk 1 above)
Comment (shashwatS22) [reply subgraphs/gmx-forks/src/modules/token.ts:26]: These were extra comments which I made while working, removed them.

File: subgraphs/gmx-forks/protocols/gmx/config/deployments/gmx-arbitrum/configurations.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "network": "arbitrum-one",
+  "protocolName": "GMX",
+  "protocolSlug": "gmx",
+  "protocolId": "0x321f653eed006ad1c29d174e17d96351bde22649",
```
Comment (dmelotik) [inline subgraphs/gmx-forks/protocols/gmx/config/deployments/gmx-arbitrum/configurations.json:5]: This needs to be updated, see the top I made for reasoning

**Comment by dmelotik** [comment]
@shashwatS22 do you know why the tvl on gmx changed slightly with the upgrade? Theoretically it should be the same, right?

https://github.com/messari/subgraphs/issues/2143#issuecomment-1582630842

**Comment by shashwatS22** [comment]
> Reference

There could be a slight change because we changed the price lib version with the upgrade. (for adding support for tokens used in mummy finance(MLP and MMY token)). Except for this, there is no other work change from the previous version.

**Comment by ishraq8** [comment]
Took a look at the TVL, the newer version has slightly higher numbers, but those numbers match the gmx analytics. Not sure what the discrepency is but the data matches the numbers from GMX. From a QA perspective this should be ok, but unsure where the difference is from. 

Screen Shot 2023-06-13 at 8 52 52 AM
Screen Shot 2023-06-13 at 8 53 02 AM
Screen Shot 2023-06-13 at 8 59 48 AM

**Comment by dmelotik** [comment]
Really nice work @shashwatS22 ! I am close to ready to merge this. I noticed one last thing, can you add:
- a readme for the forks similar to aave forks or compound forks?
- A readme for each individual protocol, similar style to the other fork repos

Once that is done, we will be ready to merge

**Comment by shashwatS22** [comment]
> Really nice work @shashwatS22 ! I am close to ready to merge this. I noticed one last thing, can you add:
> 
> * a readme for the forks similar to aave forks or compound forks?
> * A readme for each individual protocol, similar style to the other fork repos
> 
> Once that is done, we will be ready to merge

Thanks! I'll add the Readme and let you know.

**Comment by dmelotik** [comment]
Also, you can remove `gmx` and basically just copy the readme from there!

**Review by dmelotik — APPROVED** [review]

**Pull request merged**